hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/ExternalConditionEval.java

```diff
@@ -47,16 +54,33 @@
     @JsonInclude(Include.NON_NULL)
     private String value;
 
+    @ApiModelProperty(value = "Event value used for dataId.",
+            position = 2)
+    @JsonInclude(Include.NON_NULL)
+    private Event event;
+
+    @JsonIgnore
+    private transient TypeValue typeValue;
+
     public ExternalConditionEval() {
         super(Type.EXTERNAL, false, 0, null);
         this.condition = null;
         this.value = null;
+        this.typeValue = TypeValue.DATA;
+    }
+
+    public ExternalConditionEval(ExternalCondition condition, Event event) {
+        super(Type.EXTERNAL, condition.match(event.getText()), event.getCtime(), event.getContext());
+        this.condition = condition;
+        this.event = event;
+        this.typeValue = TypeValue.DATA;
```

Comment:
If you keep the typeValue field I think this should be EVENT.  But as I said above, I don't think you need it.